scheduler/server/load_based_sched_alg.go

```diff
@@ -0,0 +1,781 @@
+	// 100 - sum of % to the class with largest % (this eliminates rounding errors, forcing the
+	// % to add up to 100%)
+	totalPercents := 0
+	firstClass := true
+	for _, className := range lbs.classByDescLoadPct {
+		if firstClass {
+			firstClass = false
+			continue
+		}
+		jc := lbs.jobClasses[className]
+		jc.tempNormalizedPct = int(math.Floor(float64(jc.tempEntitlement) * 100.0 / float64(entitlementTotal)))
+		totalPercents += jc.tempNormalizedPct
+	}
+	lbs.jobClasses[lbs.classByDescLoadPct[0]].tempNormalizedPct = 100 - totalPercents
+}
+
+// computeLoanPercents as orig load %'s normalized to exclude classes that don't have waiting tasks
+func (lbs *LoadBasedAlg) computeLoanPercents() {
+	// get the sum of all the original load pcts for classes that have waiting tasks
+	pctsTotal := 0
+	for _, jc := range lbs.jobClasses {
+		if jc.numWaitingTasks > 0 {
+			pctsTotal += jc.origTargetLoadPct
+		}
+	}
+
+	if pctsTotal == 0 {
+		return
+	}
+
+	// compute the % for all but the class with the largest %.  Add up all computed %s and assign
+	// 100 - sum of % to the class with the largest % from the range (this eliminates rounding errors, forcing the
+	// sum or % to go to 100%)
+	totalPercents := 0
+	firstClass := true
+	firstClassName := ""
+	for _, className := range lbs.classByDescLoadPct {
+		jc := lbs.jobClasses[className]
+		if jc.numWaitingTasks > 0 {
+			if firstClass {
+				firstClass = false
+				firstClassName = className
+				continue
+			}
+			jc.tempNormalizedPct = int(math.Floor(float64(jc.origTargetLoadPct) * 100.0 / float64(pctsTotal)))
+			totalPercents += jc.tempNormalizedPct
+		} else {
+			jc.tempNormalizedPct = 0
+		}
+	}
+	lbs.jobClasses[firstClassName].tempNormalizedPct = 100 - totalPercents
+}
+
+// buildTaskStartList builds the list of tasks to be started for each jobClass.
+func (lbs *LoadBasedAlg) buildTaskStartList() []*taskState {
+	tasks := []*taskState{}
+	for _, jc := range lbs.jobClasses {
+		if jc.numTasksToStart <= 0 {
+			continue
+		}
+		classTasks := lbs.getTasksToStartForJobClass(jc)
+		tasks = append(tasks, classTasks...)
+	}
+	return tasks
+}
+
+// getTasksToStartForJobClass get the tasks to start list for a given jobClass.  The jobClass's numTasksToStart
+// field will contain the number of tasks to start for this job class.  The jobClass's jobsByNumRunningTasks is
+// a map from an integer value (number of tasks running) to the list of jobs with that number of tasks running
+// For a given jobClass, we start adding tasks from the jobs with the least number of tasks running.
+// (Note: when a task is started for a job, the job is moved to the ‘next’ bin and placed at the end of that bin’s job list.)
+func (lbs *LoadBasedAlg) getTasksToStartForJobClass(jc *jobClass) []*taskState {
+	tasks := []*taskState{}
+
+	startingTaskCnt := 0
+	// work our way through the class's jobs, starting with jobs with the least number of running tasks,
```

Comment:
lowest pct of running tasks would allow jobs with many tasks to starve jobs with few tasks.  I think trying to have an equal number of tasks running for each job in a class is a better solution.